### DISCUSSION TITLE
bs prefix in data attributes is really needed ?

### DISCUSSION THREAD

**Discussion by zorn-v** [twbs/bootstrap#32527] [General]
I think it like "I think I can break prev layouts with bootstrap, and because I can - I do it".
Do you really think that some "external lib" does not belong to bootstrap/jquery/etc ?
Even so, I think EVERYONE name its attributes like `data-app-something`
Its like python2 -> python3. But at least in python there were reasons.

# Suggestion 

Handle data attributes WITH and WITHOUT `bs-` prefix.
Yes, some more code, but less negative.
Maybe show some deprecation message in console etc.

**Comment by XhmikosR** [comment]
There is reason https://github.com/twbs/bootstrap/issues/29479

**Reply by zorn-v** [reply]
I see only "YEA it will be great" without analize.
Even if "it great" it no so hard to implement both selectors and log deprecation instead of "nothings work"

**Reply by rohit2sharma95** [reply]
@zorn-v There are JavaScript libraries that use same data attributes as Bootstrap. For example: jQuery UI
This was done to prevent conflicts between external libraries (or your own code) and Bootstrap. If Bootstrap still supports data attributes without bs namespace, conflict remains like before.

**Reply by zorn-v** [reply]
Some one use bootstrap and  jQuery UI together ? Strange people.
Generally I am for renaming, but maybe do it more gradually. Say show deprecation in 5.0 and drop in 5.1 with message like "data-toggle is not supported anymore" etc. if there is some attributes in layout.
Something like
```js
if (document.querySelector('[data-toggle]')) {
  console.warn('Bootstrap 5 does not support "data-toggle" attribute anymore. Use "data-bs-toggle"')
}
```
And drop this checks in next version

**Reply by rohit2sharma95** [reply]
What if user is using bs in data attributes used for Bootstrap and `data-toggle` is used in user's own code, for that should Bootstrap warn? :thinking:
